src/bp/nlu-core/engine.ts

```diff
@@ -292,9 +292,17 @@ export default class Engine implements NLU.Engine {
   }
 
   async detectLanguage(text: string, modelsByLang: _.Dictionary<string>): Promise<string> {
+    trainDebug(`Detecting language for input: "${text}"`)
+
     const predictorsByLang = _.mapValues(modelsByLang, id => this.modelsById.get(id)?.predictors)
     if (!this._dictionnaryIsFilled(predictorsByLang)) {
-      throw new Error(`one of models is not loaded: ${modelsByLang}`)
+      const missingLangs = _(predictorsByLang)
+        .toPairs()
+        .filter(([lang, pred]) => _.isUndefined(pred))
+        .fromPairs()
```

Comment:
Don't need fromPairs here.

You can .map('0') right away

Reply:
:+1:

Reply:
same here